sklearn/preprocessing/_encoders.py

```diff
@@ -846,3 +850,247 @@ def inverse_transform(self, X):
+                             % (indices.shape[0] - 1, n_features))
+
+        # We clip those ordinal features of X that are greater than categories_
+        # using mask if self.handle_greater is "clip".
+        # This means, the row_indices and col_indices corresponding to the
+        # greater ordinal feature are all filled with ones.
+        mask = (X >= self.categories_).ravel()
+        if np.any(mask):
+            if self.handle_greater == 'warn':
+                warnings.warn("Found %d feature values which exceeds "
+                              "n_categories during transform, clipping them."
+                              % np.count_nonzero(mask))
+            elif self.handle_greater == 'error':
+                raise ValueError("handle_greater='error' but found %d feature"
+                                 " values which exceeds n_categories during "
+                                 "transform." % np.count_nonzero(mask))
+
+        X_ceil = np.where(mask.reshape(X.shape), self.categories_ - 1, X)
+        column_start = np.tile(indices[:-1], n_samples)
+        column_end = (indices[:-1] + X_ceil).ravel()
+        column_indices = np.hstack([np.arange(s, e) for s, e
+                                   in zip(column_start, column_end)])
+        row_indices = np.repeat(np.arange(n_samples, dtype=np.int32),
+                                X_ceil.sum(axis=1))
+        data = np.ones(X_ceil.ravel().sum())
+        out = sparse.csr_matrix((data, (row_indices, column_indices)),
+                                shape=(n_samples, indices[-1]),
+                                dtype=self.dtype)
+
+        return out if self.sparse else out.toarray()
+
+    def inverse_transform(self, X):
+        """Convert the data back to the original representation.
+
+        Parameters
+        ----------
+        X : array-like or sparse matrix of shape \
+            (n_samples, n_encoded_features)
+            The transformed data.
+
+        Returns
+        -------
+        X_tr : array-like of shape (n_samples, n_features)
+            Inverse transformed array.
+        """
+
+        check_is_fitted(self, 'categories_')
+        X = check_array(X, accept_sparse='csr', ensure_min_features=0)
+
+        n_samples, _ = X.shape
+        n_features = len(self.categories_)
+        n_encoded_features = self.feature_indices_[-1]
+
+        # validate shape of passed X
+        msg = ("Shape of the passed X data is not correct. Expected {0} "
+               "columns, got {1}.")
+        if X.shape[1] != n_encoded_features:
+            raise ValueError(msg.format(n_encoded_features, X.shape[1]))
+
+        # return float dtype, even though it will contain int values
+        X_tr = np.zeros((n_samples, n_features), dtype=np.float)
+
+        j = 0
+        for i in range(n_features):
+            n_columns = self.categories_[i] - 1
+
+            sub = X[:, j:j + n_columns]
+
+            categories = sub.sum(axis=1).ravel()
```

Comment:
I don't think sum is the only obvious way to do this. Alternatives include summing after binarising (i.e. >0) before summing; another is to look for the position of the first zero. I don't think the sum of the raw values is acceptable, as the content being inverse-transformed will often not be binary. Whatever strategy should be documented and tested.

Reply:
I don't understand this comment @jnothman. You seem to imply that the values of `sub` aren't binary but they must be, since this **is** `inverse_transform()`.

We could indeed look for the position of the first zero: this is strictly equivalent to summing since all the previous values would be ones.

Using `sum` is the most convenient tool I found, because the code needs to be compatible with both numpy arrays **and** sparse matrices. Note that `sub.sum(...)` is also used in the `inverse_tranform()` of `OneHotEncoder` for similar a purpose.

Of course, `inverse_transform()` is tested in `tests/test_encoders.py`.

Reply:
Thank you for the details, I assumed `inverse_transform` was only suppose to accept the output of `transform()`.

Just to confirm that I correctly understand the use-case, you are thinking about cases where `UnaryEncoder` is followed by a lossy transformer in a pipeline, right?

In 57538f0 I changed the strategy to look for the first zero. This treats all non-zero entries as 1, and accepts non binary matrices. LMK if that's OK.

-----

If you don't mind I'd also appreciate your input on this https://github.com/scikit-learn/scikit-learn/pull/12893#discussion_r245341039. @qinhanmin2014 and I think that `UnaryEncoder` might be more user friendly if it accepted the same inputs as `OrdinalEncoder`.

Reply:
I'm okay with that strategy, thanks!

Hmmm... inverse_transform should be added to the glossary